master-firmware/src/trace/trace.c

```diff
@@ -0,0 +1,145 @@
+#include "trace.h"
+#include <stddef.h>
+#include <string.h>
+
+#ifdef _CHIBIOS_RT_
```

Comment:
Are you sure that condition is true ? This is defined in `ch.h`, not by a compiler define. I think your issue with timestamps might come from this.